packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/SchemaDrivenAnnotationEditor.tsx

```diff
@@ -259,127 +266,125 @@ export function SchemaDrivenAnnotationEditor(
+            formData={formData}
+            onChange={({ formData }) => {
+              if (onChange) {
+                onChange(formData)
+              }
+              setFormData(formData)
+              setValidationError(undefined)
+            }}
+            onBlur={() => {
+              setFormData(
+                // Clean the formData onBlur to remove null values that we will need to strip before submission
+                // This will ensure that the user gets accurate validation information since the data will match what the backend will receive
+                cleanFormData(
+                  formData,
+                  // Don't remove null values in arrays--the fields will disappear, which the user probably does not want
+                  false,
+                ),
+              )
+            }}
+            onSubmit={({ formData, errors }, event) => {
+              event.preventDefault()
+              if (errors && errors.length > 0) {
                 setValidationError(errors)
-                if (validationError && entityId) {
-                  setShowConfirmation(true)
-                }
-              }}
-              widgets={{
-                TextWidget: TextWidget,
-                DateTimeWidget: DateTimeWidget,
-                SelectWidget: SelectWidget,
-                CheckboxWidget: BooleanWidget,
-              }}
-            >
-              {submissionError && showSubmissionError && (
-                <Alert severity="error" sx={{ my: 2 }}>
-                  Annotations could not be updated: {submissionError.reason}
-                </Alert>
-              )}
-              {!formRefFromParent && (
-                <>
-                  <Divider sx={{ my: 2 }} />
-                  <Box
-                    display="flex"
-                    justifyContent="space-between"
-                    sx={{ gridRowStart: 5 }}
-                  >
-                    <ConfirmationButtons
-                      hasCancelButton={onCancel !== undefined}
-                      onCancel={() => {
-                        onCancel && onCancel()
-                      }}
-                      onConfirm={() => {
-                        ref.current!.formElement.current.requestSubmit()
-                      }}
-                      confirmButtonProps={{
-                        children: entityId ? 'Save' : 'Validate',
-                      }}
-                    />
-                  </Box>
-                </>
-              )}
-            </Form>
-          )}
+              }
+              setShowSubmissionError(false)
+              setFormData(formData)
+              submitChangedEntity()
+            }}
+            onError={(errors: RJSFValidationError[]) => {
+              // invoked when submit is clicked and there are client-side validation errors
+              setValidationError(errors)
+              if ((validationError || liveValidate) && entityId) {
```

Comment:
If `liveValidate` is off, and the annotations are invalid against the schema, clicking this once triggers validation, then clicking it again opens a confirmation modal to save the invalid annotations.

With this change, if `liveValidate` is on, then the user can already see the errors, so we should just 'skip' the first click and go straight to the confirmation modal.